src/shell/js/composeui-fdc3/src/ComposeUIListener.ts

```diff
@@ -0,0 +1,71 @@
+/* 
+ *  Morgan Stanley makes this available to you under the Apache License,
+ *  Version 2.0 (the "License"). You may obtain a copy of the License at
+ *       http://www.apache.org/licenses/LICENSE-2.0.
+ *  See the NOTICE file distributed with this work for additional information
+ *  regarding copyright ownership. Unless required by applicable law or agreed
+ *  to in writing, software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ *  or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ *  
+ */
+
+import { Context, ContextHandler, Listener } from "@finos/fdc3";
+import { MessageRouter, TopicMessage } from "@morgan-stanley/composeui-messaging-client";
+import { Unsubscribable } from "rxjs";
+import { Fdc3ChannelMessageRouterMessage } from "./Fdc3ChannelMessageRouterMessage";
+
+export class ComposeUIListener implements Listener {
+    private messageRouterClient: MessageRouter;
+    private unsubscribable?: Unsubscribable;
+    private handler?: ContextHandler;
+    private channelId?: string;
+    private contextType?: string;
+    private isSubscribed: boolean = true;
+
+    constructor (messageRouterClient: MessageRouter, handler?: ContextHandler, channelId?: string, contextType?: string) {
+        this.messageRouterClient = messageRouterClient;
+        this.handler = handler;
+
+        if(channelId != null) { //topic: "composeui/fdc3/v2.0/userchannels/{name}/"
+            this.channelId = channelId;
+        }
+        
+        this.contextType = contextType;
+    }
+
+    public async subscribe(channel: string): Promise<void> { //"composeui/fdc3/v2.0/userchannels/{name}/broadcast"
```

Comment:
Can someone call `subscribe` more than once? E.g.: for different channels. In that case we'd need more than one `unsubscribable`

Reply:
In the current implementation no, as per I assume we would like to export just the DesktopAgent implementation, which would call the listener's subscribe method with the suffix "broadcast" in the addContextListener method. I am not really sure if that's the right approach for the fdc3 though.